plugins/woocommerce-admin/client/tasks/fills/purchase.js

```diff
@@ -47,31 +47,38 @@ const PurchaseTaskItem = () => {
 		installedPlugins
 	);
 	const { remainingProducts } = groupedProducts;
+	const DefaultTaskItem = defaultTaskItem;
 
+	return (
+		<>
+			<DefaultTaskItem
+				onClick={ () => {
+					if ( remainingProducts.length ) {
+						toggleCartModal();
+					}
+				} }
+			/>
+			{ cartModalOpen && (
+				<CartModal
+					onClose={ () => toggleCartModal() }
+					onClickPurchaseLater={ () => toggleCartModal() }
+				/>
+			) }
+		</>
+	);
+};
+
+const PurchaseTaskItemFill = () => {
 	return (
 		<WooOnboardingTaskListItem id="purchase">
-			{ ( { defaultTaskItem: DefaultTaskItem } ) => (
-				<>
-					<DefaultTaskItem
-						onClick={ () => {
-							if ( remainingProducts.length ) {
-								toggleCartModal();
-							}
-						} }
-					/>
-					{ cartModalOpen && (
-						<CartModal
-							onClose={ () => toggleCartModal() }
-							onClickPurchaseLater={ () => toggleCartModal() }
-						/>
-					) }
-				</>
+			{ ( { defaultTaskItem } ) => (
+				<PurchaseTaskItem defaultTaskItem={ defaultTaskItem } />
 			) }
 		</WooOnboardingTaskListItem>
```

Comment:
This is a SlotFill which was retrieving all of it's data without being displayed yet. So I moved that logic out in it's own component, that way we don't make any unnecessary requests when the homescreen first loads.